controllers/topology/desired_state.go

```diff
@@ -156,11 +156,16 @@ func computeControlPlane(_ context.Context, s *scope.Scope, infrastructureMachin
 
 		// Compute the labels and annotations to be applied to ControlPlane machines.
 		// We merge the labels and annotations from topology and ClusterClass.
+		// We also add the cluster-name and the topology owned labels.
 		topologyMetadata := s.Blueprint.Topology.ControlPlane.Metadata
 		clusterClassMetadata := s.Blueprint.ClusterClass.Spec.ControlPlane.Metadata
+
+		machineLabels := mergeMap(topologyMetadata.Labels, clusterClassMetadata.Labels)
+		machineLabels[clusterv1.ClusterLabelName] = cluster.Name
+		machineLabels[clusterv1.ClusterTopologyOwnedLabel] = ""
 		if err := contract.ControlPlane().MachineTemplate().Metadata().Set(controlPlane,
 			&clusterv1.ObjectMeta{
-				Labels:      mergeMap(topologyMetadata.Labels, clusterClassMetadata.Labels),
+				Labels:      machineLabels,
```

Comment:
Are these changes necessary for this PR?

Reply:
Move to this PR: https://github.com/kubernetes-sigs/cluster-api/pull/5211